src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -122,16 +136,23 @@ void remove_virtual_functionst::remove_virtual_function(
   goto_programt new_code_gotos;
 
   exprt this_expr=code.arguments()[0];
-  // If necessary, cast to the last candidate function to
-  // get the object's clsid. By the structure of get_functions,
-  // this is the parent of all other classes under consideration.
-  const auto &base_classid=functions.back().class_id;
-  const auto &base_function_symbol=functions.back().symbol_expr;
-  symbol_typet suggested_type(base_classid);
+  const auto &last_function_symbol=functions.back().symbol_expr;
+
+  // If necessary, cast to java.lang.Object to extract the class identifier:
+  symbol_typet suggested_type("java::java.lang.Object");
```

Comment:
NOOOOOOOOOOOOOOOOOOOOOO. My apologies for shouting. This is _not_ the Java front-end. This is code in goto-programs.

Reply:
My understanding was the C++ frontend didn't use this, and this was de facto lowering a Java-only construct, but I can try some better trickery if not.

Reply:
@kroening might be the only one to fully clarify re C++. I don't know for sure, but if it truly is only useful with Java then this entire code should go in the java_bytecode folder. I think it _should_ be useful with C++, hence would ask for a more creative solution. If such a solution cannot be found, then the code should be moved to the appropriate folder.